### PR TITLE
Feat/text style customization

### PR DESCRIPTION
**Pull request by tanany365**
* Allow client code to customize text style of the phone number input field

### CODE REVIEW

**Comment by tanany365** [comment]
Hello @hbmartin 
A potential usecase is changing font size, weight, family and other non-color related props of the text field. TBH, in my own code base I used it to increase font size of the phone number field because it was too small for my potential user

>  1. Does this potentially conflict with `colors`?
* Partially yes, `TextStyle` affects text color, but only limited to the input text value, on the other hand, `colors` affects , according to documentation,
>  color of the text and content (including label, placeholder, leading and trailing icons, border)

Actually `TextStyle.color` always takes presedence over `color` property, 
but in case the property `color` is defined only and no `color` property is set in `TextStyle`, 
the standalone property `color` will take presedence.

-------------------------------------------------------------------

<br /> 
<br /> 
> 2. Would it be useful to pass this style into the search field into `SearchTextField`?

Yes, I think so, 

-------------------------------------------------------------------

<br /> 
<br /> 

>3. conflicts with other files in my other PRs?

Yes, becasue they all originate from the `main` brnach and there are some inter secting files like README, so I resolve conflicts of each PR separately after merging the previous ones.

**Review by hbmartin — APPROVED** [review]
Approved pending sync with `master`

**Pull request merged**
